lib/src/util/windows/wsa.cc

```diff
@@ -40,8 +40,12 @@ namespace facter { namespace util { namespace windows {
 
     string wsa::saddress_to_string(SOCKET_ADDRESS const& addr) const
     {
-        DWORD size = INET6_ADDRSTRLEN;
-        wchar_t buffer[INET6_ADDRSTRLEN];
+        if (!addr.lpSockaddr) {
+            return {};
+        }
+
+        DWORD size = INET6_ADDRSTRLEN+1;
+        wchar_t buffer[INET6_ADDRSTRLEN+1];
```

Comment:
I'm tempted not to, because I think we're unlikely to run into this error as long as the sockaddr is initialized. It's a pretty simple structure.